tcp_check/datadog_checks/tcp_check/tcp_check.py

```diff
@@ -76,10 +77,16 @@ def addrs(self):
 
     def resolve_ips(self):
         # type: () -> None
-        self._addrs = [
-            AddrTuple(sockaddr[0], socket_type)
-            for (socket_type, _, _, _, sockaddr) in socket.getaddrinfo(self.host, self.port, 0, 0, socket.IPPROTO_TCP)
-        ]
+        if self.ipv4_only:
+            _, _, ipv4_list = socket.gethostbyname_ex(self.host)
+            self._addrs = [AddrTuple(ipv4_addr, socket.AF_INET) for ipv4_addr in ipv4_list]
+        else:
+            self._addrs = [
+                AddrTuple(sockaddr[0], socket_type)
+                for (socket_type, _, _, _, sockaddr) in socket.getaddrinfo(
+                    self.host, self.port, 0, 0, socket.IPPROTO_TCP
+                )
```

Comment:
Note:
IP6 support was introduced on https://github.com/DataDog/integrations-core/pull/11740/files#diff-e1afdae3cf81bdf12d2888a230f697bfd8eda2d5ab2e8d7267e30183986a636bR104
The first branch of this if reverts the behaviour to use `[socket.gethostbyname(self.host)]`